ocaml/xapi/create_storage.ml

```diff
@@ -75,13 +75,42 @@ let maybe_create_pbd rpc session_id sr device_config me =
   then Client.PBD.create ~rpc ~session_id ~host:me ~sR:sr ~device_config ~other_config:[]
   else List.hd pbds (* Otherwise, return the current one *)
 
-let create_storage (me: API.ref_host) rpc session_id __context : unit =
+let maybe_remove_tools_sr rpc session_id __context =
+  let srs = Db.SR.get_all ~__context in
+  let tools_srs =
+    List.filter (fun self -> Db.SR.get_is_tools_sr ~__context ~self) srs in
+  let old_srs =
+    List.filter (fun self ->
+        let other_config = Db.SR.get_other_config ~__context ~self in
+        (Db.SR.get_is_tools_sr ~__context ~self = false) &&
+        ((List.mem_assoc Xapi_globs.tools_sr_tag other_config) ||
+         (List.mem_assoc Xapi_globs.xensource_internal other_config))
+      ) srs in
+  let unplug_and_maybe_destroy sr =
+    (* Unplug me *)
+    let pbd, _ = Sm.get_my_pbd_for_sr __context sr in
+    Db.PBD.set_currently_attached ~__context ~self:pbd ~value:false;
+    (* Destroy any Tools SRs *)
+    try
+      Client.SR.forget rpc session_id sr;
+      info "Tools SR %s has been removed" (Ref.string_of sr)
+    with
+    | Api_errors.Server_error (e, _) when e = Api_errors.sr_has_pbd ->
+      info "Tools SR %s could not be removed: it is still in use by another host" (Ref.string_of sr)
+    | e ->
+      warn "Failed to destroy redundant Tools SR %s: %s"
```

Comment:
do you mean `forget` instead of `destroy` here?

Reply:
Yeah, "destroy the DB object" and "forget the SR"... it's confusing... :)